Framework/PythonInterface/plugins/algorithms/IndirectTransmission.py

```diff
@@ -114,7 +114,7 @@ def PyExec(self):
 
         logger.notice('Analyser : ' + analyser + reflection + ' with energy = ' + str(efixed))
 
-        SetSampleMaterial(InputWorkspace=workspace, ChemicalFormula=formula)
+        SetSampleMaterial(InputWorkspace=workspace, ChemicalFormula=formula, SampleNumberDensity=density)
```

Comment:
Should we do something similar here and get rid of the `MaterialBuilder` in here too and just switch on the property name? This also avoids doing the calculation twice?